### DISCUSSION TITLE
0.5.0 Development Status

### DISCUSSION THREAD

**Discussion by z-huang** [z-huang/music#421] [Announcements]
As some features are hard or troublesome to add in the current UI framework, I'm rewriting the UI in Jetpack Compose (and also learning). Once the migration is usable, I'll open a branch for you to try. There's no ETA, and the migration may take a long time, please be patient.🙂

NOTE: To keep this thread clean, please only make replies to my comments and don't make a separate comment.

**Comment by z-huang** [comment]
- [x] Dynamic theme #159

<img src="https://user-images.githubusercontent.com/39427017/201469039-ea43ad8c-528c-4423-9796-c23ba7d60a9b.jpg" width=150 /> <img src="https://user-images.githubusercontent.com/39427017/201469040-e88118eb-6cfe-43e8-b48f-aa8bcc973e1c.jpg" width=150 /> <img src="https://user-images.githubusercontent.com/39427017/201471584-29784e20-89c5-40e6-a188-ed827178dcf1.jpg" width=150 /> <img src="https://user-images.githubusercontent.com/39427017/201471541-cab113b5-bd1d-4f9a-8d62-077a7740a1de.jpg" width=150 />

I'm in love with Compose... It's much easier to build UI with Compose.

**Reply by owenconnorz** [reply]
For dark theme could it be as dark as this dude 
<img width=150 src="https://user-images.githubusercontent.com/96526481/201470846-b7bf8b5d-d0e1-4109-8bfe-9c86769c6fe8.jpg" />

**Reply by owenconnorz** [reply]
Also cannot wait to try this 😍

**Reply by z-huang** [reply]
Added dark mode screenshot.

**Reply by owenconnorz** [reply]
Freaking awsome 😍 looks so good man how easy is it to migrate the app dude

**Reply by z-huang** [reply]
It's easy to build UI, but I still have to rewrite all things to Compose.

**Reply by NeeRaj-2401** [reply]
> I still have to rewrite all things to Compose.

just being curious 😅
alpha or beta release when ?

**Reply by z-huang** [reply]
I have no ETA.

**Comment by z-huang** [comment]
- [x] Now playing indicator #208

The indicator appears not only in the queue screen, but everywhere.

https://user-images.githubusercontent.com/39427017/202383702-a2f04f7b-2cbb-417d-8f3e-89e918906842.mp4

**Reply by owenconnorz** [reply]
Niceeee man 😁

**Reply by Ekki2015** [reply]
Awsome 👊

**Comment by z-huang** [comment]
Update:
I'm making the top bar design to achieve the following:
1. Better search UI/UX: Currently, you need to be in the home tab to search online and other tabs to search local. That's not a good UX, and I want to switch online/local by one button in the same search screen.
2. Better artist page: immersive artist banner
3. Switching between normal mode and action mode

The design will resemble the toolbar in Gmail app. If you have a better design or idea, please comment below.
<img width=300 src="https://user-images.githubusercontent.com/39427017/201463049-03442018-146c-4a42-8275-6a410b41ed6d.jpg" />

Also, I'm time-constrained recently, so the migration process may take a long time...

**Reply by owenconnorz** [reply]
Is it possible to see the branch and once is ready then add the action after so it's a like sneak peak off 👀

**Reply by z-huang** [reply]
It's meaningless to make the branch public when it's not fully functional yet.

**Reply by ghost** [reply]
![Screenshot_20221121_093910_Samsung Notes](https://user-images.githubusercontent.com/117234693/203038853-6ba01652-6d50-4a05-b088-72f7476eb320.jpg)
![Screenshot_20221121_093643_Samsung Notes](https://user-images.githubusercontent.com/117234693/203038862-901c22e7-d81e-45bc-b2ff-9add92974716.jpg)
Maybe something like this?

**Reply by z-huang** [reply]
Yeah, probably a settings icon or a hamburger icon.

**Reply by z-huang** [reply]
What do you think?

https://user-images.githubusercontent.com/39427017/203234159-4f0a90cb-3741-4b87-82bc-d6e74aaa045f.mp4

**Reply by siggi1984** [reply]
👍

**Reply by siggi1984** [reply]
> What do you think?
> 
>  Screenrecorder-2022-11-22-13-42-29-232.mp4

Nice

**Reply by ghost** [reply]
Looks great

**Reply by Ekki2015** [reply]
Very Nice 👊

**Reply by xmha97** [reply]
> ![Screenshot_20221121_093910_Samsung Notes](https://user-images.githubusercontent.com/117234693/203038853-6ba01652-6d50-4a05-b088-72f7476eb320.jpg)
> ![Screenshot_20221121_093643_Samsung Notes](https://user-images.githubusercontent.com/117234693/203038862-901c22e7-d81e-45bc-b2ff-9add92974716.jpg)
> Maybe something like this?

Like Google apps, add settings button, here:
<img width=250 src="https://user-images.githubusercontent.com/23485114/203771378-23d46a47-d86a-401e-a32d-e22c68e270f9.jpg" />

**Reply by NeeRaj-2401** [reply]
@z-huang any new update ? regarding the progress

**Reply by z-huang** [reply]
The development pace is steady. This is the immersive artist screen. Working toward the playlist screen...

<img width=200 src="https://user-images.githubusercontent.com/39427017/204137575-cd50c4bd-c2e5-44ab-a7a8-f87506c6778f.jpg" />

**Reply by NeeRaj-2401** [reply]
Looks amazing 🙌 thanx

**Reply by siggi1984** [reply]
Nice. Keep up the good work, the app is just mega 👊

**Reply by cat3na** [reply]
For the search UI: what about including a toggle like this to switch between library only/YT music results? This wouldn't have to change anything about the main screen's search bar. 

<img width=200 src="https://user-images.githubusercontent.com/118707951/204200246-e34d768c-9d7b-4172-a40e-726b787a5851.png" />

The text in the search bar could change to indicate what section you're searching.

<img width=200 src="https://user-images.githubusercontent.com/118707951/204200272-b632ab73-eecf-4520-a4ee-525f459b0d26.png" />

Alternative wording for the toggle:

<img width=200 src="https://user-images.githubusercontent.com/118707951/204200318-76c8abe0-49fa-43a9-8767-53dd4d23748b.png" />

**Reply by NeeRaj-2401** [reply]
Rather a toggle below search bar what about this 

![image](https://user-images.githubusercontent.com/88180887/204201831-61fc708d-1791-4943-ac0f-8084c2d1acbc.png) 
(design is adopted from somewhere else so modification according to app would look much better)

obviously including current voice search option and then the filters options ( YT and Library as filter options )as @cat3na  mentioned

Also, keep YT results as default and IF user wants library searches then he/she can use the toggle to switch the search functionality

**Reply by siggi1984** [reply]
> Rather a toggle below search bar what about this
> 
> ![image](https://user-images.githubusercontent.com/88180887/204201831-61fc708d-1791-4943-ac0f-8084c2d1acbc.png) (design is adopted from somewhere else so modification according to app would look much better)
> 
> obviously including current voice search option and then the filters options ( YT and Library as filter options )as @cat3na mentioned
> 
> Also, keep YT results as default and IF user wants library searches then he/she can use the toggle to switch the search functionali

I personally like the idea of NeeRaj-2401 the best

**Reply by ghost** [reply]
> Rather a toggle below search bar what about this
> 
> ![image](https://user-images.githubusercontent.com/88180887/204201831-61fc708d-1791-4943-ac0f-8084c2d1acbc.png) (design is adopted from somewhere else so modification according to app would look much better)
> 
> obviously including current voice search option and then the filters options ( YT and Library as filter options )as @cat3na mentioned
> 
> Also, keep YT results as default and IF user wants library searches then he/she can use the toggle to switch the search functionality

I think this idea is great but I don't know if it would look good on a phone screen, I'm worried it would look too small

**Reply by NeeRaj-2401** [reply]
![image](https://user-images.githubusercontent.com/88180887/204243895-4fbc2b04-53e8-4868-b8af-ce12edeec0db.png)

all i wanted to say was ... just a filter button next to voice search icon would look much better, minimal and clean

so the search idea suggested by @cat3na will work and it will look good too

**Reply by z-huang** [reply]
I prefer @NeeRaj-2401 's idea. Adding a icon button to switch between local and online would look minimal and doesn't take up much space.
Also, I want to remove the mic button to get more space. The feature is already in the IME.

**Reply by cat3na** [reply]
I agree, and I think a single button to switch would be perfect once the voice search button is gone. I would be a little concerned about how discoverable it is without a label, though. What kind of icon are you thinking of?

**Reply by NeeRaj-2401** [reply]
> What kind of icon are you thinking of?

some kind of filter or search filter icon would work ?
and just like other icons tap and hold will show the info of button

**Reply by z-huang** [reply]
Online
<img width=200 src="https://user-images.githubusercontent.com/39427017/204997961-e409ec03-8fd7-4d5b-a0a2-d8e914edfeab.png" />
Local
<img width=200 src="https://user-images.githubusercontent.com/39427017/204998123-cff1ec4f-81e1-4a57-8db0-cca17e591ac5.png" />

**Reply by NeeRaj-2401** [reply]
Amazing 🙌

**Reply by cat3na** [reply]
This looks great. Would it be possible to retain the "Search YouTube Music..." vs "Search library..." text in the search box, as it is currently? This might give an extra indication of why the button is there.

**Reply by z-huang** [reply]
Yes, of course.

**Comment by z-huang** [comment]
New design bottom menu sheet

<img width=200 src="https://user-images.githubusercontent.com/118707951/204984971-62791081-85fc-4095-b670-a71aa88bc081.png" />

**Reply by NeeRaj-2401** [reply]
@z-huang  how about list view with 2 columns ? with icon and names besides
 
 **GRID ORDER :**
start radio  -------. play next
add to playlist --- add to queue
edit --------------- download
view artist -------- view album
share -------------- refetch
----- delete ----------

all these each option in  one-one container with icon and name besides
imagine this... i guess it will look organized and appealing ...

**Reply by z-huang** [reply]
But I think list view with too many items will make it hard to find the action you want. Grid view is easier to memorize the position of every action and the space of each button is bigger. I also make similar actions in the same row.

**Reply by NeeRaj-2401** [reply]
> But I think list view with too many items will make it hard to find the action you want. Grid view is easier to memorize the position of every action. I also make similar actions in the same row.

thats why i also kept relevant actions besides in row
for e.g. 
view artist and album
add to playlist and add to queue
edit and download

it will be become habitual over the period

**Reply by NeeRaj-2401** [reply]
your idea and implementation are also good ... 
lets wait for the community response 
if they prefer your implementation then it will be good to go

**Reply by z-huang** [reply]
"Start radio", "Play next" and "Add to queue" are grouped together and need 3 columns."View artist", "View album" are grouped together and need 2 columns. Therefore, a fixed column list can't display them well.

**Reply by NeeRaj-2401** [reply]
prototype of my idea
many tweaks would required by your end 
<img src="https://user-images.githubusercontent.com/88180887/204979193-f2e5c0e2-461e-49ce-bc2b-eab7c2963f8c.png" width="25%">

**Reply by cat3na** [reply]
The grid layout feels a bit disorienting... I'm not sure if it's the amount of white space around all the grid items, or if it's the offset on the Artist/Album row, but it feels difficult to read. 

I like NeeRaj's suggestion, it seems to follow a logical layout coming from other music players and avoids feeling too cluttered.

My other suggestion would be retaining the grid layout with the bottom-right corner empty, so for example:

<img width=200 src="https://user-images.githubusercontent.com/118707951/204984971-62791081-85fc-4095-b670-a71aa88bc081.png" />

I rearranged the buttons slightly to try and help the eye distinguish them, I don't know if there are restrictions on this (you mentioned grouping?)

**Reply by z-huang** [reply]
@cat3na This looks better. It also solved the problem when some songs have no related album or artists.

**Reply by Sumitmishra903** [reply]
> Album

Ok sir

**Reply by owenconnorz** [reply]
When can we expect a debug version so we can test out as looking forward to it 😁

**Reply by z-huang** [reply]
The good news is that I'm on winter vacation now, so I have more time recently. I'll try to make it public as soon as possible.

**Reply by owenconnorz** [reply]
Thanks @z-huang 👌

**Reply by yant0** [reply]
Can we also have the option to reareange the buttons? Since it looks just like a grid and rearranging them to ones need is definitely a plus.

edit :Or maybe even change how many rows & column? Because imo. I like a tighter design like neeraj proposed as to the wide and pretty much invisible "material deisgn" look of the buttons

**Reply by z-huang** [reply]
> Can we also have the option to rearrange the buttons?

No. The positions of the buttons are designed carefully.

> Or maybe even change how many rows & column?

How about trying the new design first and share what you think is the drawback of this design? (When the preview is out)

**Comment by z-huang** [comment]
- [x] In-app volume control #454

The seek bar is really an ios design.

https://user-images.githubusercontent.com/39427017/210380743-9b7c3af3-9fd0-4086-bf56-27595d54d33d.mp4

(I think I should post updates more often to let you know the development status...)

**Reply by NeeRaj-2401** [reply]
Looks amazing, will this be triggered by volume buttons too?

**Reply by z-huang** [reply]
This is independent of the system volume. So, if you press the volume buttons, the seekbar progress won't change.

**Reply by ivan-avalos** [reply]
I think the bar should be a bit thinner.

**Reply by z-huang** [reply]
Being thick makes more space for your finger to scroll.

**Reply by jacobel640** [reply]
genius!!!

**Reply by owenconnorz** [reply]
Would this make the music louder example if my device volume was on max and that bar was on max volume would it be louder than current version we have

**Reply by z-huang** [reply]
No, 100% volume is the default volume.

**Reply by nikhilCad** [reply]
Innertune was always too loud for me for some reason, hope this feature helps

**Comment by z-huang** [comment]
Rewriting the YouTube Music library again to make specific data classes for each page. (Painful process)

**Reply by airon90** [reply]
Will we have to refetch data again or will nothing change for end user?

**Reply by z-huang** [reply]
Nothing for user to do.

**Comment by z-huang** [comment]
Rewriting repository class.

**Comment by z-huang** [comment]
Question:
You know that you can import or add a playlist to library. Do you really use the "add to library" feature? Or do you think the "sync" feature from ViMusic is better?

**Reply by NeeRaj-2401** [reply]
> do you think the "sync" feature from ViMusic is better?

This

**Reply by owenconnorz** [reply]
I always use the library section and add to library but if u do it sync by sync that won't be a issue because whatever syncs that I don't want  I can just remove

**Reply by cat3na** [reply]
I think I would usually "import playlist," because I like being able to edit the list in-app, but people who rely on online playlists (ones that get updated on YouTube) probably "add to library" more often? Either way, when I add songs/albums/etc to library it's partly just to create a convenient shortcut to them, because I'll forget the music that I could listen to if it's not right in front of my face.

I'm not really sure what the sync feature from ViMusic does. Is it like adding to library, but it refetches the list automatically?

**Reply by z-huang** [reply]
> I'm not really sure what the sync feature from ViMusic does. Is it like adding to library, but it refetches the list automatically?

It's a button to delete all songs from the playlist and import again.

**Reply by NeeRaj-2401** [reply]
> It's a button to delete all songs from the playlist and import again

Is there any way to implement some sort of bookmark (always automatically up to date) feature for playlists..

**Comment by z-huang** [comment]
Finally, branch `compose` is out! Sorry to keep you waiting so long. Please give it a try and report bugs in issues. Remember to make backup when changing version! However, it will still take some time to make it stable.

- [x] Download feature
- [x] Richer menu items
- [x] History #332
- [x] Stats
- [x] Share to InnerTune

The new home screen has only two sections - most played and new release albums. When history is implemented, I'll replace most played with quick picks. If you want other sections, please comment below.

**Reply by NeeRaj-2401** [reply]
Finally the wait is over 😅 thanx for ur hardwork.

**Reply by nikhilCad** [reply]
No accent colour for 0.5.0?

**Reply by nikhilCad** [reply]
weird issue with ime text having different colour.
<img width=200 src="https://user-images.githubusercontent.com/47214394/217813400-e4e5be1b-c767-46a2-a602-7c97c57c3c34.png" />

Note: In Light mode both show correct dark colour

**Reply by z-huang** [reply]
> No accent colour for 0.5.0?

Default color is blue, and dynamic color is always on.

**Reply by z-huang** [reply]
> weird issue with ime text having different colour.

I can't reproduce. However, sometimes when I switch dark mode, text colors are not changing.

**Reply by nikhilCad** [reply]
Oh so the colour changes with song, nice

**Reply by airon90** [reply]
```compose``` branch misses Refetch feature too, @z-huang

**Reply by NeeRaj-2401** [reply]
Is only selected bottom nav is planned in compose too? Which were previously contributed by bnyro(LibreTube dev).

**Reply by owenconnorz** [reply]
Can recommend playlists, partying and all hits be added to the home screen please 😁

**Reply by radical4ever** [reply]
There are new components to MD3 i would love to see the carrousel one on the home screen page and the library page too!!

**Reply by NeeRaj-2401** [reply]
@z-huang  i just have a question
vimusic contains almost every necessary features and have 2.57mb apk file. After installation for first time its occupies approximately 7-8mb.

And innerTune by default its 17-18mb. 
So code cleanup & ditching unnecessary Library would help in to reduce innerTube apk size or any other way...?

Is it possible to reduce innerTune's apk size? In your opinion

**Reply by z-huang** [reply]
Debug apks are bigger. I tested that a release apk of `compose` took 6.3MB, which is acceptable.

**Reply by owenconnorz** [reply]
@z-huang is swipe to refresh able to be added back for the home screen dude

**Reply by z-huang** [reply]
No need to have that. Most played songs are automatically updated. New release albums doesn't update frequently.

**Reply by owenconnorz** [reply]
OK what about say like your offline then u go back online because the homescreen doesn't load unless restart app or unless there's away for the homescreen to load once the app sees there's a connection again 😊

**Reply by owenconnorz** [reply]
Hi @z-huang are the quick picks meant to show stuff as I'm getting a blank screen a part from new albums

**Reply by z-huang** [reply]
Try bdf01fb885eb740d7aa566ac8c60b332289ae5ff and play some songs, and it should show up.

**Reply by owenconnorz** [reply]
Yeah worked after listening after music and then refreshed the app quick picks showed

**Reply by Dragonfly-30** [reply]
Sorry, if it's a dumb question but how can I download beta?

**Reply by owenconnorz** [reply]
https://github.com/z-huang/InnerTune/suites/11210808329/artifacts/573476943

**Reply by NeeRaj-2401** [reply]
@z-huang menu items (such as play next, add to queue etc)are missing from **album screen**

**Reply by z-huang** [reply]
> Can recommend playlists, partying and all hits be added to the home screen please 😁

It's hard to determine which playlists to be shown in the screen, and I find YouTube's home playlists not helpful for me. Therefore, not planned.

**Comment by seasidesongs** [comment]
I have restored my library to the new beta version but when I try and play a song it says source error 🤔🤔

**Reply by siggi1984** [reply]
I had that too. You have to remove the downloads in the stable version and then backup.

Then restore them in the beta version.
That fixed the error for me.

**Comment by TheTrashCrafter** [comment]
Is this supposed to have hw acceleration?

**Reply by z-huang** [reply]
Do you mean compose? Yes.

**Reply by TheTrashCrafter** [reply]
No, I meant if the test builds use hw acceleration. Because it feels really choppy on my Pixel 6a (GrapheneOS)

**Reply by TheTrashCrafter** [reply]
Is that going to be fixed in the full release?

**Reply by z-huang** [reply]
Compose has a different rendering process than the original UI framework. Choppy animations are probably because the code is causing too much recomposition. As a beginner in compose, it's easier to make such mistake. I'll try to analyze of the root causing composable functions and optimize them. It's not related to hardware acceleration.

**Reply by TheTrashCrafter** [reply]
Ok, thank's for your response. The Compose Branch looks awesome btw.

**Reply by ivan-avalos** [reply]
It actually surprised me how performant the app continues to be, even with Compose! I haven't noticed laggy or slow animations, compared to other apps I use that also migrated to Compose. You're doing such a great job! Keep it up!

**Reply by TheTrashCrafter** [reply]
@ivan-avalos Hmm.... Maybe it's because of my Pixel or because of Graphene.

**Reply by ivan-avalos** [reply]
@TheTrashCrafter Any Pixel should be a lot more performant than my phone, so maybe Graphene is the issue. Are you running a release APK or a debug APK?

**Reply by TheTrashCrafter** [reply]
@ivan-avalos The Debug one... Yeahhhh, i probably should've known that compose debug apks are horribly slow.

**Reply by TheTrashCrafter** [reply]
@ivan-avalos I switched back to stock now. Everything works like a charm now, even with the debug apk.

**Comment by radical4ever** [comment]
Is thre going to be any widgets on this next release? Like youtube music's one?
<img width=200 src="https://user-images.githubusercontent.com/83252104/218282474-d5b556d9-ed5d-42d2-8ea9-cb4652f6e6ec.png" />

**Reply by z-huang** [reply]
Not planned.

**Reply by xmha97** [reply]
@z-huang When you publish the new version?

**Reply by NeeRaj-2401** [reply]
@xmha97 its under active development. u can try [debug version](https://github.com/z-huang/InnerTune/suites/11247967151/artifacts/576149325)

some bugs and [missing things](https://github.com/z-huang/InnerTune/discussions/421#discussioncomment-4917947) are to be completed yet. Keep your eyes on [this thread](https://github.com/z-huang/InnerTune/discussions/421#discussioncomment-4917947) to know more.
![image](https://user-images.githubusercontent.com/88180887/222052831-a147e243-6b2f-4373-ab94-58e0a58d76e2.png)

**Reply by z-huang** [reply]
There's no ETA, and I'm quite busy because the semester has started. It will be likely to take a long time.

**Reply by xmha97** [reply]
Ok, thanks for your reply.

**Comment by EisregenHaha** [comment]
Edit: just saw its your semester so everything is fine after all 😅 glad to hear and good luck with ur studies

Hey bro is everything fine? This project has been quite inactive so thats why im asking. I am not expecting you to work 24/7 i am just wondering if all is good with u cuz we havent heard from u in so long. Best wishes

**Reply by owenconnorz** [reply]
@z-huang are you back 😁😁

**Reply by siggi1984** [reply]
That's mega that the development here continues. The best music stream
 app.

Much more important is his study that everything runs super there.

**Reply by owenconnorz** [reply]
Definitely the development of this app is awsome and very missed obviously we all understand that his studies are more important but seeing the 1st commit in months is exciting lol 😆

**Reply by EisregenHaha** [reply]
> Edit: just saw its your semester so everything is fine after all 😅 glad to hear and good luck with ur studies
> 
> Hey bro is everything fine? This project has been quite inactive so thats why im asking. I am not expecting you to work 24/7 i am just wondering if all is good with u cuz we havent heard from u in so long. Best wishes

Holy shi he's back 🔥

**Comment by owenconnorz** [comment]
@z-huang I was thinking maybe you could make the icons centre with the titles underneath and maybe add a option to remove the name of each 3 buttons and use just the icons.
Didn't add this to the issues due to being more for the development progress.
Thanks man and hope your doing well 🙏 

<img src="https://github.com/z-huang/InnerTune/assets/96526481/1e8e33f7-d458-47aa-a529-d9388180766f" width=500 />

**Reply by EisregenHaha** [reply]
<img src="https://github.com/z-huang/InnerTune/assets/98850034/eb34e58e-fcef-42ce-9401-5b57f9768fb6" width=500 />

You should lower your dpi so that the font isnt as big as my grammys phone lmao, it looks normal for me

**Reply by owenconnorz** [reply]
I have a z fold 3 so font being reduced would be tiny lol

**Comment by z-huang** [comment]
The semester finally passed, and I’ll continue the development of this project. After being absent for a few months, I got a different goal on this project: It isn’t aimed to be as powerful as possible. It should be minimal but useful, with all must-need features. Therefore, some “advanced” feature request will either be labeled as “contribution only” or be closed as not planned (don’t be angry). I’ll go through all issues and reconsider whether it’s too “advanced” or not. Some decisions will be changed, and I apologize for that. After all, I don’t have infinite time on this project and I will only focus on what’s most important.

During these days, I got many issues. I’m so thankful that you tell me the bugs and suggest new features. However, some issues aren’t following the form. Just pasting the error message without any explanation isn’t helpful at all. You know that there is a checklist in the issue form, and I hope that you check these things carefully before ticking the checkbox. Failing to do so will waste my time, slow down the development, and eventually delay the release of new version.

Since this project is not my whole life and I have other things to do, I’ve set up ways to sponsor this project, which can be a stimulation for me to spend more time on this. If you find this app useful, please consider being a sponsor.😍

**Reply by EisregenHaha** [reply]
Awesome man, thanks also for implementing the toggle for dynamic theme i requested.

**Reply by siggi1984** [reply]
So in my opinion it is already a mega app. Many thanks for that.
The works quite reliably and also fulfills its purpose. The next version 0.5.0 already looks mega. Since it is already a mega app I would just focus on the functions that do not yet run properly that these work. Otherwise, many many thanks for your hard work.

**Reply by Vishcot** [reply]
Hi, Thanks btw, It is the best project i have ever come across to... i am so much excited for the newer version and I want to try the debug version 0.5.0 but i don't know how to compile( not a programmer), is there any chance or anyone who can directly share the early debug build .apk version, or will i be needed to wait till the stable release?

**Reply by EisregenHaha** [reply]
Just download it from the releases tab. Click on the latest entry with checkmark, so successful build, and then on the app.zip or whatever the name is

**Reply by Vishcot** [reply]
> Just download it from the releases tab. Click on the latest entry with checkmark, so successful build, and then on the app.zip or whatever the name is

It has a 0.4.4 version only, which i am already using, I want to try the latest build developer version - 0.5.0..

**Reply by EisregenHaha** [reply]
Sorry i didnt mean releases tab, i meant actions tab. The rest of the instructions are right though

**Reply by Vishcot** [reply]
Thankyou ✌️ it worked!

**Reply by EisregenHaha** [reply]
Np

**Reply by GuiSousa135** [reply]
Very good, but don't leave out the important and "advanced" functions, mainly related to downloading, lyrics, widgets, in short, this type of detail makes all the difference so that this application remains the best of all music apps. I am very interested in making donations to contribute to the advancement of the project. Do you have an estimate of when the new version will be released?

**Reply by z-huang** [reply]
@GuiSousa135 If everything goes well, I'll release 0.5.0 before September, when the next semester starts.

**Reply by mmtxrc** [reply]
with the swift pace things are going at i feel like uninstalling spotify every passing day. desperately awaiting 0.5.0.

**Reply by mmtxrc** [reply]
AAAAAAAAAAA i did it AAAAAAAAAAA
innertune+antennapod

**Comment by z-huang** [comment]
What are your thoughts about the quick picks? Do the suggestions match your taste?

**Reply by Vishcot** [reply]
Far better than what we got in 0.4.0 version and what youtube music has already given in its App, but still not much useful..i dont usually use that feature..

**Reply by JPGuillemin** [reply]
It's great. Enough for me. No need to connect to YTM account

**Reply by radical4ever** [reply]
I would just need my ytm account to sync my playlistand maybe my taste history. It could be a good move to have that option

**Reply by owenconnorz** [reply]
Having my account on here would be awsome and hope we still get the login.

**Reply by ghost** [reply]
The quick pics seem good and more personalised than 0.4.4. Sure not as specific as Spotify, but honestly I'm impressed in comparison to all other ''privacy respecting'' streaming apps. Did a good job I'd say

**Reply by z-huang** [reply]
Is it better now? a10b5a7a7828ccf53a279b7b0ad627d3ac94b302

**Reply by JPGuillemin** [reply]
I really love it !!!

**Reply by TheTrashCrafter** [reply]
It seems good, but I don’t really use it.

**Comment by TheTrashCrafter** [comment]
Here are some suggestions:
- Better animations when you open the player, like in 0.4.1
- circumvent age restrictions as there is no way of signing in to proof that you're 18+, this already worked in 0.3.3 and earlier when it was still using NewPipe Extractor.
- Make the navbar always show like in non-compose versions

Apart from that I really have no idea on what could be improved as it's already a great app. Thank you for putting so much effort into the app, for me it's already better than Spotify and the only thing that keeps me using Apple Music is the second point.

**Reply by z-huang** [reply]
> Better animations when you open the player

Too difficult to implement in Jetpack Compose.

> circumvent age restrictions

Done.

> Make the navbar always show like in non-compose versions

Not planned.

**Reply by TheTrashCrafter** [reply]
Hi, thanks for your reply, but sadly it didn't really fix the issue. When I try to play age restricted songs it gives me unknown error instead of playing. Is it impossible nowadays?

**Reply by owenconnorz** [reply]
Apparently it's piped api that is the issue as it uses piped extractor for the age restriction bypass but it's not working

**Reply by TheTrashCrafter** [reply]
@owencz1998 Well, it works using 0.3.3 (last version using NewPipe Extractor). But if the problem is piped, then we can't do anything about it I guess.

**Reply by owenconnorz** [reply]
I dunno I tested the song I wanted on another app and it worked just bot this app.

**Reply by TheTrashCrafter** [reply]
@owencz1998 ViMusic doesn't work anymore either, so I guess it's not the devs fault

**Reply by owenconnorz** [reply]
Oh no I wasn't blaming the only thing I can think of is we would need a login system to be able for it to work

**Reply by TheTrashCrafter** [reply]
@owencz1998 That would be the perfect solution, but I can already hear some people complaining about Privacy...

**Reply by owenconnorz** [reply]
Tho this would be the best feature lol and helpful in other ways lol

**Comment by z-huang** [comment]
I think it's time to release 0.5.0. Other enhancements will be added in later versions. You can try this stable build: [app-release.apk.zip](https://github.com/z-huang/InnerTune/files/11955248/app-release.apk.zip) (Remove the `.zip` suffix and install). If you find any bugs, please create an issue. Also, is the performance good enough in stable builds?

**Reply by mmtxrc** [reply]
its perfect ✨ more than good enough to be stable
imma be using it for longer tho. to see for any bugs. nitpicky ones. found none as of now
love love 💟

**Reply by ghost** [reply]
Tested this out for about 30 mins and seems stable. 
Performance is good enough on a Pixel 7a. Good job, loving it.

edit: dont forget to try and add this app to f-droid!

**Reply by mmtxrc** [reply]
oh and uhm is the icon not supposed to support monet? its only blue and that too an off tone.
![Screenshot_2023-07-05-17-42-43-991_com miui home](https://github.com/z-huang/InnerTune/assets/94820934/9818b51f-c069-4739-9598-99e37625a5f2)

**Reply by z-huang** [reply]
InnerTune supports monochrome icon.

**Reply by GuiSousa135** [reply]
I found some issues/bugs in update 0.5.0:

1 - Shuffer button in the music tab should disappear when scrolling to the end of the page (to release the "three dots button" of the last song).

2 - Should have a option to search for specific songs within the selected playlist tab.

3 - Multiselection of songs within a playlist or in the music tab is no longer available.

As for the issue of instability and fluidity, it's very good, and the new visuals are incredible.

**Reply by TheTrashCrafter** [reply]
Age restriction bypass still isn't working, even though the whatever.social servers are back online.

**Reply by z-huang** [reply]
Some songs work: https://www.youtube.com/watch?v=x8VYWazR5mE. It depends on Piped API.

**Comment by LouieEj** [comment]
Liking a song and adding a song to library should be merged into one. There is never a time that I want to Like a song (adding it to Liked Songs playlist) but not add it to my library, and vice versa.

I think this will make the app easier to understand and to use, as currently if I want to properly "Like" a song like in other apps I have to press both the Like button and the Add To Library button.

If this is done, I believe the Songs screen could simply be for Liked Songs, and the "Liked Songs" playlist is just shown in the songs screen instead of in the playlists screen (or it could be in both).

**Reply by NeeRaj-2401** [reply]
Yeah this point makes sense, liked songs getting considered in all over such as in songs tabs, artists etc
And removing add it to library

Or vice versa

**Reply by mmtxrc** [reply]
no no no hang on
the solution to ur problem is only to introduce the behaviour of adding the song to library simultaneously as u like it.
merging liked songs with add to library is not a good idea. 
keeping them seperated enables better organisation of songs. library has music of ur taste and liked songs contains the absolutely loved ones.
auto-add to library when clicking the heart it is.

**Reply by LouieEj** [reply]
I understand your point, but I believe it just makes it confusing. Even with your suggestion, users will be a bit confused as to why 'Liking' a song boths adds the song to library and to Liked Playlist, whereas 'Adding to Library' only adds to library (the latter makes sense, but the difference in what they do is confusing as there is an overlap in functionality for two different buttons)

Your point is a good one, but that can already be achieved with playlists - if users want to have songs that they absolutely love, they can put those songs in a playlist they create. And then any song that they listen to and they like can just be 'Liked'. There is no need for an 'Add to Library' button, it just makes users confused (like myself and friends who have tried the app).

**Comment by mmtxrc** [comment]
a weird behaviour. a youtube video after a yt song. open lyrics changes akbum cover and idk swiping back takes me to start of queue.  

https://github.com/z-huang/InnerTune/assets/94820934/73bf5b34-6de4-400e-975b-9a4f3192fb50
